opm/simulators/thresholdPressures.hpp

```diff
@@ -187,7 +187,7 @@ void computeMaxDp(std::map<std::pair<int, int>, double>& maxDp,
             int pvtRegionIdx = pvtRegion[cellIdx];
 
             double T = initialState.temperature()[cellIdx];
-            double p = initialState.pressure()[cellIdx];
+            double p = phasePressure[wpos][cellIdx];
```

Comment:
I am not sure about this, but it seems to be a correct fix. However, I'd appreciate if @totto82 could evaluate the effect of this on model 2 (which is the only case we have that uses this feature) before we merge it. In any case it *will* change the solution, and should therefore not be done lightly.